### PR TITLE
{devel}[foss/2020b] PyTorch v1.7.1 w/ Python 3.8.6

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    ('PyYAML', '5.3.1'),
+    ('MPFR', '4.1.0'),
+    ('GMP', '6.2.0'),
+    ('numactl', '2.0.13'),
+    ('FFmpeg', '4.3.1'),
+    ('Pillow', '8.0.1'),
+]
+
+# default CUDA compute capabilities to use (override via --cuda-compute-capabilities)
+cuda_compute_capabilities = ['3.5', '3.7', '5.2', '6.0', '6.1', '7.0', '7.2', '7.5']
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb:34]: Agh, yes. indeed.

File: easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+]
+patches = [
+    'PyTorch-1.6.0_fix-test-dataloader-fixed-affinity.patch',
+    'PyTorch-1.7.0_fix_altivec_defines.patch',
+    'PyTorch-1.7.0_fix_test_DistributedDataParallel.patch',
+    'PyTorch-1.7.0_fix-fbgemm-not-implemented-issue.patch',
+    'PyTorch-1.7.0_avoid-nan-in-test-torch.patch',
+    'PyTorch-1.7.0_increase-distributed-test-timeout.patch',
+    'PyTorch-1.7.0_disable-dev-shm-test.patch',
+    'PyTorch-1.7.1_el8_ppc64le.patch',
```
Comment (branfosj) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb:242]: This patch is not needed for a `foss` build. (It is also not needed when we build with CUDA 11).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3123.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/90bc854998adfd87c6dc716b49dd5d62 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 15 out of 15 (2 easyconfigs in total)
nuc.lan - Linux Fedora 32, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.8.6
See https://gist.github.com/91b43950e668e3d03742df548743897b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2629.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/f122a36c35ffc7cfdc4f802e7c1e3191 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/ceee73dcfa8e066e1ff4e901485fc972 for a full test report.

File: easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,308 @@
+name = 'PyTorch'
+version = '1.7.1'
+
+homepage = 'https://pytorch.org/'
+description = """Tensors and Dynamic neural networks in Python with strong GPU acceleration.
```
Comment (Micket) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb:5]: Should we really say "GPU acceleration" in the description?

Comment (boegel) [reply easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb:5]: It's the standard description taken from https://github.com/pytorch/pytorch...

I understand your point though, people may assume they're using a GPU-accelerated installation.

Then again, is using different descriptions in different easyconfigs maintainable?

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 14 out of 14 (2 easyconfigs in total)
node250.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/f28f0814b292f1b5a4cf6f7dccbb6b49 for a full test report.

File: easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb (hunk 3 above)
Comment (Micket) [reply easybuild/easyconfigs/p/PyTorch/PyTorch-1.7.1-foss-2020b.eb:5]: well, alright, i don't think users really read these anyway. An alternative would be to switch for both of them, with the description from pytorch.org "An open source machine learning framework that accelerates the path from research prototyping to production deployment.". Or not.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @boegel !

**Pull request merged**